### ISSUE TITLE
npm deprecated warning for few packages

### ISSUE THREAD

**Issue by ansAGujar** [angular/angular#57274]
### Which @angular/* package(s) are the source of the bug?

common

### Is this a regression?

Yes

### Description

I have created the new project with angular and do npm install. I am getting below warning.

npm install
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated read-package-json@7.0.1: This package is no longer supported. Please use @npmcli/package-json instead.
npm warn deprecated backbone-undo@0.2.6: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info. 
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported

package.json file:

{
  "name": "test-project",
  "version": "0.0.1",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "watch": "ng build --watch --configuration development",
    "test": "ng test --watch=false --browsers=ChromeHeadless",
    "lint": "ng lint",
    "pretty-quick": "pretty-quick --staged",
    "prepare": "husky",
    "compodoc": "compodoc -p tsconfig.json",
    "compodoc-serve": "compodoc -s"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^17.1.0",
    "@angular/common": "^17.1.0",
    "@angular/compiler": "^17.1.0",
    "@angular/core": "^17.1.0",
    "@angular/elements": "^17.3.12",
    "@angular/forms": "^17.1.0",
    "@angular/platform-browser": "^17.1.0",
    "@angular/platform-browser-dynamic": "^17.1.0",
    "@angular/router": "^17.1.0",
    "angular-font-awesome": "^3.1.2",
    "bootstrap": "^5.3.3",
    "grapesjs": "^0.21.12",
    "grapesjs-preset-webpage": "^1.0.3",
    "husky": "^9.1.1",
    "rxjs": "~7.8.0",
    "tslib": "^2.3.0",
    "zone.js": "~0.14.3"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^17.1.2",
    "@angular/cli": "^17.1.2",
    "@angular/compiler-cli": "^17.1.0",
    "@types/jasmine": "~5.1.0",
    "angular-eslint": "18.1.0",
    "eslint": "^9.6.0",
    "jasmine-core": "~5.1.0",
    "karma": "~6.4.0",
    "karma-chrome-launcher": "~3.2.0",
    "karma-coverage": "~2.2.0",
    "karma-jasmine": "~5.1.0",
    "karma-jasmine-html-reporter": "~2.1.0",
    "prettier": "^3.3.3",
    "pretty-quick": "^4.0.0",
    "typescript": "~5.3.2",
    "typescript-eslint": "8.0.0-alpha.38"
  }
}

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 17.1.2
Node: 18.13.0
Package Manager: npm 8.19.3
OS: win32 x64

Angular: undefined
...

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.1701.2 (cli-only)
@angular-devkit/core         17.1.2 (cli-only)
@angular-devkit/schematics   17.1.2 (cli-only)
@schematics/angular          17.1.2 (cli-only)
```

**Comment by JeanMeche** [comment]
Hi,
You're repporting dependencies from a project that isn't up-to-date (v17 while the current version is v18) and also isn't a bare project with several other dependencies. 

Feel free to re-open if you can report deprecated dependencies on a bare v18 project.

**Comment by ansAGujar** [comment]
@JeanMeche I have created the new project with v18, still getting similar warning for npm packages.

package.json

{
  "name": "demo-test-1",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "watch": "ng build --watch --configuration development",
    "test": "ng test"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^18.1.0",
    "@angular/common": "^18.1.0",
    "@angular/compiler": "^18.1.0",
    "@angular/core": "^18.1.0",
    "@angular/forms": "^18.1.0",
    "@angular/platform-browser": "^18.1.0",
    "@angular/platform-browser-dynamic": "^18.1.0",
    "@angular/router": "^18.1.0",
    "rxjs": "~7.8.0",
    "tslib": "^2.3.0",
    "zone.js": "~0.14.3"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^18.1.3",
    "@angular/cli": "^18.1.3",
    "@angular/compiler-cli": "^18.1.0",
    "@types/jasmine": "~5.1.0",
    "jasmine-core": "~5.1.0",
    "karma": "~6.4.0",
    "karma-chrome-launcher": "~3.2.0",
    "karma-coverage": "~2.2.0",
    "karma-jasmine": "~5.1.0",
    "karma-jasmine-html-reporter": "~2.1.0",
    "typescript": "~5.5.2"
  }
}


![image](https://github.com/user-attachments/assets/ce837aa3-3efe-4d2e-977b-3aa11e5230a4)

**Comment by JeanMeche** [comment]
Those deprecated dependencies are pulled by `karma` which pulls a old version of `glob`. 
This isn't actionnable on our side.

**Comment by ansAGujar** [comment]
Thank you!!

**Comment by dzocespd** [comment]
@ansAGujar this issue happened to me today, and while googling I came up to the issue you opened. 
I got this issue when tried to init Angular new app while using node version v20.12.2.
I resolved it by upgrading to LTS node version v20.16.0 and removed npm cache. 

I hope this helps 😄
